lib/ohai/mixin/network_helper.rb

```diff
@@ -47,7 +53,26 @@ def hex_to_dec_netmask(netmask)
       # server), and the method should return the hostname and not the IP address.
       #
       def canonicalize_hostname(hostname)
-        Addrinfo.getaddrinfo(hostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first.canonname
+        ai = Addrinfo
+          .getaddrinfo(hostname, nil, nil, nil, nil, Socket::AI_CANONNAME)
+          .first
+
+        canonname = ai&.canonname
+        # use canonname if it's an FQDN
+        # This API is preferred as it never gives us an IP address for broken DNS
+        # (see https://github.com/chef/ohai/pull/1705)
+        # However, we have found that Windows hosts that are not joined to a domain
+        # can return a non-qualified hostname)
+        return canonname unless ip?(canonname)
```

Comment:
this isn't what you want, this will have the same bug. If this returns "myhostname", you'll return it, and that's not what you want. 

We KNOW this won't be an IP address since `canonname` only returns names, so we just want to check it has a dot in it.